keydata_test.go

```diff
@@ -448,70 +600,130 @@ func (s *keyDataSuite) checkKeyDataJSONAuthModeNone(c *C, keyData *KeyData, crea
+func (s *keyDataSuite) checkKeyDataJSONAuthModePassphrase(c *C, keyData *KeyData, creationParams *KeyWithPassphraseParams, nmodels int, passphrase string, kdfOpts *KDFOptions) {
 	w := makeMockKeyDataWriter()
 	c.Check(keyData.WriteAtomic(w), IsNil)
 
 	s.checkKeyDataJSONFromReaderAuthModePassphrase(c, w.Reader(), creationParams, nmodels, passphrase, kdfOpts)
 }
 
 type testKeyPayloadData struct {
-	key    DiskUnlockKey
-	auxKey AuxiliaryKey
+	primary PrimaryKey
+	unique  []byte
+}
+
+func marshalASN1(c *C, primary PrimaryKey, unique []byte) []byte {
+	builder := cryptobyte.NewBuilder(nil)
+
+	builder.AddASN1(cryptobyte_asn1.SEQUENCE, func(b *cryptobyte.Builder) { // ProtectedKeys ::= SEQUENCE {
+		b.AddASN1OctetString(primary) // primary OCTETSTRING
+		b.AddASN1OctetString(unique)  // unique OCTETSTRING
+	})
+
+	b, err := builder.Bytes()
+	c.Assert(err, IsNil)
+	return b
 }
 
 func (s *keyDataSuite) testKeyPayload(c *C, data *testKeyPayloadData) {
-	payload := MarshalKeys(data.key, data.auxKey)
+	payload := marshalASN1(c, data.primary, data.unique)
 
-	key, auxKey, err := payload.Unmarshal()
+	pk, err := UnmarshalProtectedKeys(payload)
 	c.Check(err, IsNil)
-	c.Check(key, DeepEquals, data.key)
-	c.Check(auxKey, DeepEquals, data.auxKey)
+
+	dj := new(bytes.Buffer)
+	json.NewEncoder(dj).Encode(data)
+
+	pkj := new(bytes.Buffer)
+	json.NewEncoder(pkj).Encode(pk)
+
+	c.Check(dj.Bytes(), DeepEquals, pkj.Bytes())
 }
 
 func (s *keyDataSuite) TestKeyPayload1(c *C) {
-	key, auxKey := s.newKeyDataKeys(c, 32, 32)
+	primary := s.newPrimaryKey(c, 32)
+	// Not really a primary key just using the same method
+	// to generate a random value of the same size
+	unique := s.newPrimaryKey(c, 32)
 
 	s.testKeyPayload(c, &testKeyPayloadData{
-		key:    key,
-		auxKey: auxKey})
+		primary: primary,
+		unique:  unique})
 }
 
 func (s *keyDataSuite) TestKeyPayload2(c *C) {
-	key, auxKey := s.newKeyDataKeys(c, 64, 32)
+	primary := s.newPrimaryKey(c, 64)
+	unique := s.newPrimaryKey(c, 32)
 
 	s.testKeyPayload(c, &testKeyPayloadData{
-		key:    key,
-		auxKey: auxKey})
+		primary: primary,
+		unique:  unique})
 }
 
 func (s *keyDataSuite) TestKeyPayload3(c *C) {
-	key, _ := s.newKeyDataKeys(c, 32, 0)
+	primary := s.newPrimaryKey(c, 32)
 
 	s.testKeyPayload(c, &testKeyPayloadData{
-		key: key})
+		primary: primary})
 }
 
 func (s *keyDataSuite) TestKeyPayloadUnmarshalInvalid1(c *C) {
-	payload := make(KeyPayload, 66)
+	payload := make([]byte, 66)
 	for i := range payload {
 		payload[i] = 0xff
 	}
 
-	key, auxKey, err := payload.Unmarshal()
-	c.Check(err, ErrorMatches, "EOF")
-	c.Check(key, IsNil)
-	c.Check(auxKey, IsNil)
+	if s.Version == 1 {
+		key, auxKey, err := UnmarshalV1KeyPayload(payload)
+		c.Check(err, ErrorMatches, "EOF")
+		c.Check(key, IsNil)
+		c.Check(auxKey, IsNil)
+	} else {
+		pk, err := UnmarshalProtectedKeys(payload)
+		c.Check(err, ErrorMatches, "malformed input")
+		c.Check(pk, IsNil)
+	}
```

Comment:
Isn't this test only ever called with `s.Version == 2`. In any case, this tests 2 separate functions - it should really be 2 separate tests.

Reply:
This also doesn't exist anymore as it was and it is now split in 2 tests:
`TestKeyPayloadUnmarshalInvalid1` and the new one `TestLegacyKeyPayloadUnmarshalInvalid1`